src/sql-parser/src/parser.rs

```diff
@@ -5653,14 +5653,27 @@ impl<'a> Parser<'a> {
         };
         let as_of = self.parse_optional_as_of()?;
         let up_to = self.parse_optional_up_to()?;
+        let envelope = if self.parse_keyword(ENVELOPE) {
+            Some(self.parse_subscribe_envelope()?)
+        } else {
+            None
+        };
         Ok(Statement::Subscribe(SubscribeStatement {
             relation,
             options,
             as_of,
             up_to,
+            envelope,
         }))
     }
 
+    fn parse_subscribe_envelope(&mut self) -> Result<SubscribeEnvelope, ParserError> {
+        self.expect_keyword(UPSERT)?;
+        self.expect_keyword(KEY)?;
```

Comment:
The syntax in the issue is `UPSERT (KEY a,b,c)`. Was there a reason to move KEY outside the parens? Note: I don't know why the KEY keyword is here at all. Can it just...be removed? Probably need to ask more generally as there may be a good reason.

Reply:
@benesch thought thinks that `ENVELOPE UPSERT (KEY (a, b, c))` is best. It does seem more future proof in case theres another argument we want to give ENVELOPE UPSERT